src/library_pthread.js

```diff
@@ -154,7 +154,18 @@ var LibraryPThread = {
       }
       PThread.runningWorkers = [];
     },
+#if USE_ASAN || USE_LSAN
+    freeThreadData__deps: ['$withBuiltinMalloc'],
+#endif
     freeThreadData: function(pthread) {
+#if USE_ASAN || USE_LSAN
+      // When sanitizers are enabled, free is normally instrumented to call
+      // sanitizer code that checks some things about pthreads. We are not
+      // ready for such calls, as this can be called after the sanitizers
+      // are finalized. Instead, call free directly.
+      withBuiltinMalloc(function () {
+#endif
```

Comment:
The other places that we use `withBuiltinMalloc` are not guarded with `if` like this.    Instead `withBuiltinMalloc` becomes a no-op.  So I think you can avoid the `if` checks here.

Reply:
I tried that initially, but got this exception when I compiled without sanitizers (for e.g. when running `wasm2.test_pthread_atexit`).
```
exception thrown: ReferenceError: withBuiltinMalloc is not defined
```